elastic/kibana
# [ML] Notifications indicator

## Summary

Resolves https://github.com/elastic/kibana/issues/135417

Adds a notifications indicator to the side nav menu. 

#### How it works

- When the user visits the Notifications page, the app stashes the timestamp of the latest notification visible on the screen in the local storage 
- The app polling the `.ml-notifications*` index with an interval of 1 minute for new notifications that occurred after the stored timestamp  
- If there are only notifications with an `info` level, displays the generic notification indicator 

<img width="1517" alt="image" src="https://user-images.githubusercontent.com/5236598/191020063-92a0e638-b47a-431c-b491-443167d39c81.png">

- If there are errors or warnings, it shows a counter in the side nav.  
<img width="1407" alt="image" src="https://user-images.githubusercontent.com/5236598/191029711-f0830d21-2d55-4c17-8cfe-026bbb4172b4.png">

- When the user visits the page again, sets the time bounds based on the last checked notification. 

#### How to test

- To trigger an `info` message, you can create any job or deploy model, start/stop datafeed, etc.
- To trigger an `error`, you can delete the source index of the anomaly detection job. Setting `frequency` for this job to something small should make it less time-consuming. 


### Checklist
